src/request/builder.rs

```diff
@@ -422,6 +420,39 @@ impl<B: Body> RequestBuilder<B> {
     }
 }
 
+impl<B> RequestBuilder<B> {
+    /// Inspect the properties of this request
+    pub fn inspect(&self) -> RequestInspector<'_, B> {
+        RequestInspector(self)
+    }
+}
+
+/// Allows to inspect the properties of a request before preparing it.
+#[derive(Debug)]
+pub struct RequestInspector<'a, B>(&'a RequestBuilder<B>);
```

Comment:
Should be it called `RequestBuilderInspector`? It's technically what it is, but it feels a bit Java-esque.

Reply:
The type name will probably never be spelled out in user programs, so I have no problem with changing the name. However, I do concur that `RequestBuilderInspector` is the road towards `AbstractBaseClassImplFactory` and would personally keep the simpler name.

Reply:
Yeah ok, it's not a big deal.